questions/108-medium-trim/template.ts

```diff
@@ -1 +1,11 @@
-export type Trim<S extends string> = any
+//Similar to the previois exercise, but for this one I trim left and right with the same logic to get the final result
+type Whitespace = ' ' | '\n' | '\t'
+
+type TrimLeft<S extends string> = S extends `${Whitespace}${infer Rest}`
+  ? TrimLeft<Rest>
+  : S
+type TrimRight<S extends string> = S extends `${infer Rest}${Whitespace}`
+  ? TrimRight<Rest>
```

Comment:
In this case the type is being used recursively. In order to only remove the whitespace I capture the rest of the string so that when the type is called again it only trims the whitespace